File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.java
import java.util.*;

class Solution {
    public String solution(String s, String skip, int index) {
        String answer = "";
        int n = 26 - skip.length();
        Character[] alphabet = new Character[n];
        int idx = 0;
        for (char ch = 'a'; ch <= 'z'; ch++) {
            if (!skip.contains(String.valueOf(ch))) {
                alphabet[idx] = ch;
                idx++;
            }
        }
        for (int i = 0; i < s.length(); i++) {
            int idx2 = Arrays.asList(alphabet).indexOf(s.charAt(i));
            answer = answer + alphabet[(idx2 + index) % n];
        }
        return answer;
    }
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.py
# 문자열 s의 각 알파벳을 index만큼 뒤의 알파벳으로 바꾼다.
# index만큼의 뒤의 알파벳이 z를 넘어갈 경우 다시 a로
# skip에 있는 알파벳은 제외하고 건너뛴다.
# s를 변환한 결과를 return

def solution(s, skip, index):
    answer = ''
    alphabet = []
    for i in range(26):
        tmp = chr(ord('a') + i)
        if tmp in skip:
            continue
        else:
            alphabet.append(tmp)
    for i in range(len(s)):
        print((alphabet.index(s[i]) + index) % (26 - len(skip)))
        answer += alphabet[(alphabet.index(s[i]) + index) % (26 - len(skip))]
    return answer